tests/unit/parallel/segmented_algorithms/partitioned_vector_adjacent_find.cpp

```diff
@@ -0,0 +1,77 @@
+//  Copyright (c) 2017 Ajai V George
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompadjacent_finding
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_main.hpp>
+#include <hpx/include/partitioned_vector.hpp>
+#include <hpx/include/parallel_adjacent_find.hpp>
+
+#include <hpx/util/lightweight_test.hpp>
+
+#include <cstddef>
+#include <vector>
+#include <iostream>
+///////////////////////////////////////////////////////////////////////////////
+// Define the vector types to be used.
+HPX_REGISTER_PARTITIONED_VECTOR(double);
+HPX_REGISTER_PARTITIONED_VECTOR(int);
+
+///////////////////////////////////////////////////////////////////////////////
+
+#define SIZE 64
+
+template <typename T>
+void initialize(hpx::partitioned_vector<T> & xvalues)
+{
+    T init_array[SIZE] = {1,2,3,4, 5,1,2,3, 1,5,2,3, 4,2,3,2, 1,2,3,4, 5,6,5,6,
```

Comment:
Could you add a test using the binary predicate?

Reply:
Please check the current test cases.